interfaces/Matlab_Toolbox_Revamp/Class/Kinetics.m

```diff
@@ -0,0 +1,590 @@
+classdef Kinetics < handle
+    
+    properties
+        owner
+        kin_id
+        Kc % equilibrium constant
+        Kf % forward reaction rate
+        Kr % reverse reaction rate
+        dH % enthalpy of reaction
+        dS % entropy of reaction
+        dG % gibbs free energy of reaction
+    end
+
+    methods
+        %% Kinetics class constructor
+
+        function kin = Kinetics(ph, src, id, n1, n2, n3, n4)
+            checklib;
+            % indices for bulk phases in a heterogeneous mechanism
+            inb1 = -1;
+            inb2 = -1;
+            inb3 = -1;
+            inb4 = -1;
+            if nargin == 2
+                id = '-';
+            end
+            kin.owner = 1;
+            % get the integer indices used to find the stored objects
+            % representing the phases participating in the mechanism
+            iph = ph.tp_id;
+            if nargin > 3
+                inb1 = n1.tp_id;
+                if nargin > 4
+                    inb2 = n2.tp_id;
+                    if nargin > 5
+                        inb3 = n3.tp_id;
+                        if nargin > 6
+                            inb4 = n4.tp_id;
+                        end
+                    end
+                end
+            end
+            kin.kin_id = calllib(ct, 'kin_newFromFile', src, id, ...
+                                 iph, inb1, inb2, inb3, inb4);
+        end
+        
+        %% Utility methods
+
+        function kin_clear(kin)
+            % Delete the kernel object
+            
+            checklib;
+            calllib(ct, 'kin_del', kin.kin_id);
+        end
+        
+        %% Get scalar attributes
```

Comment:
I'm not sure this comment applies to the very next method, which returns an array...